test/consistent-package-json.test.ts

```diff
@@ -0,0 +1,142 @@
+
+    // Template keys should match exactly. If updating it, remember to update tests!
+    expect(keys).toStrictEqual(KNOWN_TEMPLATE_KEYS);
+  });
+});
+
+readPackages().forEach(({packageName, packageJSON, packageJSONPath}) => {
+  // eslint-disable-next-line jest/valid-describe
+  describe(packageJSONPath, () => {
+    it('specifies Shopify as author', () => {
+      expect(packageJSON.author).toBe(packageJSONTemplate.author);
+    });
+
+    it('specifies Quilt Issues as bugs URL', () => {
+      expect(packageJSON.bugs).toStrictEqual(packageJSONTemplate.bugs);
+    });
+
+    it('specifies a dependencies', () => {
+      expect(packageJSON.dependencies).not.toStrictEqual({});
+    });
+
+    it('specifies a description', () => {
+      expect(packageJSON.description).not.toBeUndefined();
+    });
+
+    it('specifies files', () => {
+      if(packageName == 'polyfills') return; // FIXME
+
+      expect(packageJSON.files).toContain('dist/*');
+      expect(packageJSON.files).toContain('!tsconfig.tsbuildinfo');
+    });
+
+    it('specifies Quilt deep-link homepage', () => {
+      const expectedHomepage = compile(packageJSONTemplate.homepage);
+
+      expect(packageJSON.homepage).toBe(expectedHomepage);
+    });
+
+    it('specifies the MIT license', () => {
+      expect(packageJSON.license).toBe(packageJSONTemplate.license);
+    });
+
+    it('specifies the expected main', () => {
+      if(packageName === 'graphql-persisted') return; // FIXME: Address this in graphql-persisted
+
+      expect(packageJSON.main).toBe(packageJSONTemplate.main);
+    });
+
+    it('specifies name matching scope and path', () => {
+      const expectedName = compile(packageJSONTemplate.name);
+
+      expect(packageJSON.name).toBe(expectedName);
+    });
+
+    it('specifies the expected publishConfig', () => {
+      expect(packageJSON.publishConfig).toStrictEqual(
+        packageJSONTemplate.publishConfig,
+      );
+    });
+
+    it('specifies a repository deep-linking into the Quilt monorepo', () => {
+      const expectedRepository = {
+        ...packageJSONTemplate.repository,
+        directory: compile(packageJSONTemplate.repository.directory),
+      };
+
+      expect(packageJSON.repository).toStrictEqual(expectedRepository);
+    });
+
+    it('specifies scripts, including build', () => {
+      expect(packageJSON.scripts.build).toBe(packageJSONTemplate.scripts.build);
+    });
+
+    it('specifies if it has sideEffects', () => {
+      const exceptions = ['csrf-token-fetcher', 'dates', 'react-effect', 'react-google-analytics', 'useful-types']
+      if (exceptions.includes(packageName)) return; // FIXME
+
+      expect(packageJSON.sideEffects).toBe(Boolean(packageJSON.sideEffects));
+    });
+
+    it('specifies the expected types', () => {
+      if(packageName === 'graphql-persisted') return; // FIXME
+
+      expect(packageJSON.types).toBe(packageJSONTemplate.types);
+    });
+
+    it('specifies a version', () => {
```

Comment:
npm will refuse to publish without a version, so this feels over-protective.

Reply:
Fair. Should it be this file's responsibility to catch it before it gets to the publish stage though?